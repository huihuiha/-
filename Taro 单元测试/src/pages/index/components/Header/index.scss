.header-wrap {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100vw;
    height: 100px;
    background-color: beige;

    .input {
        width: 500px;
        padding: 8px;
        border: 1px solid #000;
        border-radius: 5px;
    }
}