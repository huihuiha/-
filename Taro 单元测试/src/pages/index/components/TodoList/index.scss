.todo-list-wrap {
    margin-top: 20px;
    margin: 0 auto;
    width: 100vw;
    background-color: azure;
    .item-wrap {
        display: flex;
        justify-content: center;
        align-items: center;
        border-left: burlywood;
        border-left-width: 15px;
        padding-left: 10px;
        border-left-style: solid;
        .btn {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 50px;
            width: 150px;
            font-size: 30px;
            color: #444;
        }
    }
}