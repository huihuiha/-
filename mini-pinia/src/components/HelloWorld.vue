<script setup>
import { useCounterStore } from '../store/counter';
const counter = useCounterStore();
</script>

<template>
  <h1>{{ counter.count }} --- {{ counter.doubleCount }}</h1>
  <button @click="counter.count++">++</button>
  <button @click="counter.inc()">++</button>
  <!-- 更新多个属性 -->
  <button @click="counter.$patch({ count: counter.count + 1 })">++</button>
  <button @click="counter.$patch((state) => state.count++)">++</button>
</template>

<style scoped>
.read-the-docs {
  color: #888;
}
</style>
