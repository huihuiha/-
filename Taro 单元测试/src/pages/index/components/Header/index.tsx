import { View, Input, Button } from '@tarojs/components';
import { useState } from 'react';
import './index.scss';

const Header = ({
    addItem,
}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (value) => {
        setInputValue(value);
    }

    const handleAddItem = () => {
        handleInputChange('');
        addItem({ show: true, value: inputValue });
    }

    return (
        <View className='header-wrap'>
            <Input
                className='input'
                type="text"
                placeholder='请输入todo事件'
                onInput={(e) => handleInputChange(e.detail.value)}
                value={inputValue}
            />
            <Button className='btn add-btn' onClick={handleAddItem}>添加</Button>
        </View>
    )
}
export default Header;