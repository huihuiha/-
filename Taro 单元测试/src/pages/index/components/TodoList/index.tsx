import { View, Button } from '@tarojs/components';
import { useMemo } from 'react';
import './index.scss';

const TodoList = ({
    list,
    handleDeleteItem
}) => {
    // 数量展示
    const count = useMemo(() => {
        let count = 0;
        list.forEach(item => {
            if (item.show) {
                count++
            }
        })
        return count;
    }, [list])


    return (
        <View className='todo-list-wrap'>
            {
                list.map((item, index) => (
                    item.show ? (
                        <View key={item} className='item-wrap'>
                            <View className='item' key={item.value}>{item.value}</View>
                            <Button className='btn' onClick={() => handleDeleteItem(index)}>删除</Button>
                        </View>
                    ) : ''
                ))
            }
            <View className="count">{count}</View>
        </View>
    )
}
export default TodoList;