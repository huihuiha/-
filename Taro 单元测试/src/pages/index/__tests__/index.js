// import { render, screen, fireEvent } from '@testing-library/react'
import {shallow, mount, render} from 'enzyme';
import Index from '../index';
import { Input } from '@tarojs/components';
import toJson, { mountToJson } from 'enzyme-to-json'

import renderer from 'react-test-renderer';

import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'

describe('Index 首页', () => {
  test('Index 首页样式发生变化的时候，做出提示', () => {
    const wrapper = shallow(<Index />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  test('调用Index页面的addItem方法，传入给toDoList元素数量加1', () => {
    const wrapper = shallow(<Index />);
    const header = wrapper.find('Header');
    const addFn = header.props().addItem;
    addFn({show: true, value: 'TEST测试'});
    const todoList = wrapper.find('TodoList');
    expect(todoList.props().list[0]).toEqual({show: true, value: 'TEST测试'});
  })
})