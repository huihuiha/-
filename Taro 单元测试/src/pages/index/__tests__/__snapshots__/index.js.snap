// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index 首页 Index 首页样式发生变化的时候，做出提示 1`] = `
<ForwardRef
  className="index"
>
  <Header
    addItem={[Function]}
  />
  <TodoList
    handleDeleteItem={[Function]}
    list={Array []}
  />
</ForwardRef>
`;
